asllib/Interpreter.ml

```diff
@@ -442,6 +447,11 @@ module Make (B : Backend.S) (C : Config) = struct
         | NotFound -> fatal_from e @@ Error.UndefinedIdentifier x)
         |: SemanticsRule.EVar
     (* End *)
+    | E_Binop (((BAND | BOR | IMPL) as op), e1, e2)
+      when is_simple_expr e1 && is_simple_expr e2 ->
+        let* v1 = eval_expr_sef env e1 and* v2 = eval_expr_sef env e2 in
+        let* v = B.binop op v1 v2 in
+        return_normal (v, env)
```

Comment:
I am not so sure about this "optimisation". Namely in some situation it may lead to the additional evaluation of the second expression. For instance in `e1 && e2`, when both expressions are simple and that `e1` reduces to `FALSE`. Then, without the above code, `e2` would not be evaluated at all...